### DISCUSSION TITLE
AutoMapper ProjectTo incompatible with EF5 Filtered Include

### DISCUSSION THREAD

**Discussion by ww2406** [LuckyPennySoftware/AutoMapper#3715] [NA]
### Source/destination types

```csharp
public class Blog
    {
        public int Id { get; set; }
        public List<Post> Posts { get; set; }
    }

public class BlogDTO
    {
        public int Id { get; set; }
        public List<PostDTO> Posts { get; set; }
    }

public class Post
    {
        public int Id { get; set; }
        public string Title { get; set; }
        
        public int AuthorId { get; set; }
        public Author Author { get; set; }
        
        public int BlogId { get; set; }
        public Blog Blog { get; set; }
    }

public class PostDTO
    {
        public int Id { get; set; }
        public string Title { get; set; }
        public AuthorDTO Author { get; set; }
    }

public class Author
    {
        public int Id { get; set; }
        public string Name { get; set; }
        
        public List<Post> Posts { get; set; }
    }

public class AuthorDTO
    {
        public int Id { get; set; }
        public string Name { get; set; }
    }
```

### Mapping configuration

```csharp
public MappingProfile()
        {
            CreateMap<Blog, BlogDTO>();
            CreateMap<Post, PostDTO>();
            CreateMap<Author, AuthorDTO>();
        }
```

### Version: 10.1.1
EF Core is 5.0.8

### Expected behavior

When using the ProjectTo functionality with EF Core 5's new filtered include functionality, the filtered include is not processed.

```csharp
_ctx.A.Include(x => x.B.Where(y => y.C == D)).ProjectTo<E>(_mapper.ConfigurationProvider)
```

It's expected that this returns entity A with those records from B where property C equals a value D.

### Actual behavior

In looking at the resulting SQL, the where clause is never materialized into the query--it proceeds as if the where clause is not there. If you remove the ProjectTo, the where clause is included in the query as expected.

### Steps to reproduce

Please see my sample project at https://github.com/ww2406/AutomapperFilterIncludeIssue.

**Comment by lbargaoanu** [comment]
https://github.com/AutoMapper/AutoMapper/discussions/3713#discussioncomment-1141771

**Comment by ww2406** [comment]
Got it! Can’t believe all this time I never knew that!! Learn something new everyday. Many thanks!
